bracket-push.json

```diff
@@ -0,0 +1,49 @@
+{
+  "cases": [
+    {
+      "description": "empty string",
+      "input": "",
+      "expected": true
+    },
+    {
+      "description": "unpaired brackets",
+      "input": "[[",
+      "expected": false
+    },
+    {
+      "description": "wrong ordered brackets",
+      "input": "}{",
+      "expected": false
+    },
+    {
+      "description": "paired and nested brackets",
+      "input": "([{}({}[])])",
+      "expected": true
+    },
+    {
+      "description": "unpaired and nested brackets",
+      "input": "([{])",
+      "expected": false
+    },
+    {
+      "description": "paired and wrong nested brackets",
+      "input": "[({]})",
+      "expected": false
+    },
+    {
+      "description": "paired with whitespace",
+      "input": "{ }",
+      "expected": true
+    },
+    {
+      "description": "math expression",
+      "input": "(((185 + 223.85) * 15) - 543)/2",
+      "expected": true
+    },
+    {
+      "description": "complex latex expression",
+      "input": "\right(\begin{array}{cc} \frac{1}{3} & x^2 \\end{array})",
```

Comment:
Oh! Okay. Hmm, if this should be a latex string then should it be the case that all the `\` get escaped, then? Otherwise the `\r` in `\right` is going to become a carriage return, right?